### PR TITLE
make a .rsp file for `zig clang`

### LINKED ISSUE #12419: CI failing on Windows test-standalone issue_11595
Related: #12380 #12381

```
D:\\a\\1\\s\\test\\standalone\\issue_11595\\test.c:1:1: error: unable to build C object: Unexpected
error: zigtest...
error: The following command exited with error code 1:
D:\a\1\s\build\dist\bin\zig.exe build-exe @D:\a\1\s\test\standalone\issue_11595\zig-cache\args\630588031f7c4788fc4b93c39d253b3b36b1a709cfee50eb078d1917e375c2ba 
```

### PR DESCRIPTION
**Pull request by hdorio**
I copy/pasted what was done in `std.build.LibExeObjStep.make()` for `zig build-exe` to create a .rsp file for `zig clang`.

Maybe it's good enough for now?
Let me know if I need to improve anything.

closes #12419

### CODE REVIEW

File: src/Compilation.zig [hunk 1]
```diff
@@ -3786,6 +3789,71 @@ fn updateCObject(comp: *Compilation, c_object: *CObject, c_obj_prog_node: *std.P
...
+            const input_args = argv.items[2..];
+            const output_dir = comp.local_cache_directory;
+
+            var args_arena = std.heap.ArenaAllocator.init(allocator);
+            defer args_arena.deinit();
+
+            const args_to_escape = input_args;
+            var escaped_args = try ArrayList([]const u8).initCapacity(args_arena.allocator(), args_to_escape.len);
+
+            arg_blk: for (args_to_escape) |arg| {
```
Comment (marler8997) [inline src/Compilation.zig:3811]: How about?
```zig
for (args_to_escape) |arg| {
    const escape_chars = "\\\"";
    if (std.mem.indexOfAny(u8, arg, escape_chars)) |first_escape_pos| {
        // Slow path for arguments that need to be escaped. We'll need to allocate and copy
        // ...[snip]
    } else {
        escaped_args.appendAssumeCapacity(arg); // no escaping needed so just use original argument
    };
}
```

File: src/Compilation.zig [hunk 2]
```diff
@@ -3786,6 +3789,71 @@ fn updateCObject(comp: *Compilation, c_object: *CObject, c_obj_prog_node: *std.P
...
+            const partially_quoted = try std.mem.join(allocator, "\" \"", escaped_args.items);
+            const args = try std.mem.concat(allocator, u8, &[_][]const u8{ "\"", partially_quoted, "\"" });
+
+            // Write the args to zig-cache/args/<SHA256 hash of args> to avoid conflicts with
+            // other zig build commands running in parallel.
+
+            var args_hash: [Sha256.digest_length]u8 = undefined;
+            Sha256.hash(args, &args_hash, .{});
+            var args_hex_hash: [Sha256.digest_length * 2]u8 = undefined;
+            _ = try std.fmt.bufPrint(
```
Comment (marler8997) [inline src/Compilation.zig:3838]: `cactch unrecahable` might make more sense here instead of `try`?

File: test/standalone.zig [hunk 3]
```diff
@@ -35,10 +35,6 @@ pub fn addCases(cases: *tests.StandaloneContext) void {
     if (builtin.zig_backend == .stage1) { // https://github.com/ziglang/zig/issues/12194
         cases.addBuildFile("test/standalone/issue_9812/build.zig", .{});
     }
-    if (builtin.os.tag != .windows) {
-        // https://github.com/ziglang/zig/issues/12419
-        cases.addBuildFile("test/standalone/issue_11595/build.zig", .{});
```
Comment (Vexu) [inline test/standalone.zig:40]: Why is this issue removed?

Comment (hdorio) [reply test/standalone.zig:42]: A careless mistake.

**Review by Vexu — APPROVED** [review]

**Comment by squeek502** [comment]
FYI, the usage of `realpathAlloc` here has caused a regression on Dragonfly/NetBSD/FreeBSD < 13, see https://github.com/ziglang/zig/issues/13565

**Comment by andrewrk** [comment]
Reverted in 9bcd48e40e272408879a442b21099ea5ab6b85b4 due to usage of `realpath`.

**Pull request merged**